integration-tests-scala/src/test/scala/tech/beshu/ror/integration/ClusterStateWithInternodeSslTests.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.integration
+
+import com.dimafeng.testcontainers.ForAllTestContainer
+import org.scalatest.Matchers._
+import org.scalatest.WordSpec
+import tech.beshu.ror.utils.containers.{ReadonlyRestEsCluster, ReadonlyRestEsClusterContainer}
+import tech.beshu.ror.utils.elasticsearch.ClusterStateManagerJ
+
+class ClusterStateWithInternodeSslTests extends WordSpec with ForAllTestContainer {
+
+  override val container: ReadonlyRestEsClusterContainer = ReadonlyRestEsCluster.createLocalClusterContainer(
+    name = "ROR1",
+    rorConfigFileName = "/cluster_state_internode_ssl/readonlyrest.yml",
+    numberOfInstances = 3,
+    internodeSslEnabled = true
+  )
+
+  private lazy val adminClusterStateManager = new ClusterStateManagerJ(container.nodesContainers.head.adminClient)
+
+  "Health check" should {
+    "be successful" when {
+      "internode ssl between nodes is enabled" in {
+        val response = adminClusterStateManager.healthCheck()
+
+        response.getResponseCode should be(200)
```

Comment:
should we also verify the count of cluster nodes = 3?

Reply:
I think it's not necessary.  During cluster initialization we check cluster health with configured retrying policy. If status is green (all nodes joined cluster)  we can continue with test execution, so here we are sure that cluster is in valid state and all (in this case 3) nodes are up.